src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    /// <summary>
+    /// Provides metadata about a set of types that is relevant to JSON serialization.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract partial class JsonSerializerContext
```

Comment:
Are there parts of this class defined elsewhere? Couldn't find anything at quick glance.

Reply:
The generated `JsonContext` classes derive from this class and provide the rest of the implementation.

Reply:
Right but I guess I'm asking if the class needs to be `partial`.